src/services/ReportService.ts

```diff
@@ -0,0 +1,169 @@
+import { Service } from "typedi";
+import { EntityManager, Not } from "typeorm";
+import { InjectManager } from "typeorm-typedi-extensions";
+import { UserModel } from "../models/UserModel";
+import { PostModel } from "../models/PostModel";
+import { MessageModel } from "../models/MessageModel";
+import { ReportModel } from "../models/ReportModel";
+import { ReportPostRequest, ReportProfileRequest, ReportMessageRequest } from "../types/ApiRequests";
+import Repositories, { TransactionsManager } from "../repositories";
+import { report } from "process";
+import { UuidParam } from "../api/validators/GenericRequests";
+import { NotFoundError, ForbiddenError, UnauthorizedError } from "routing-controllers";
+
+@Service()
+export class ReportService {
+  private transactions: TransactionsManager;
+
+  constructor(@InjectManager() entityManager: EntityManager) {
+    this.transactions = new TransactionsManager(entityManager);
+  }
+
+  public async getAllReports(user: UserModel): Promise<ReportModel[]> {
+    return this.transactions.readOnly(async (transactionalEntityManager) => {
+      if (!user.admin) {
+        throw new Error("User does not have permission to get all reports");
+      }
+      return Repositories.report(transactionalEntityManager).getAllReports();
+    });
+  }
+
+  public async getAllPostReports(user: UserModel): Promise<ReportModel[]> {
+    return this.transactions.readOnly(async (transactionalEntityManager) => {
+      if (!user.admin) {
+        throw new Error("User does not have permission to get all post reports");
+      }
+      return Repositories.report(transactionalEntityManager).getAllPostReports();
+    });
+  }
```

Comment:
It's probably cleaner to just replace this with middleware on the protected routes, or potentially just write a decorator which can check that the `@CurrentUser` form routing-controllers is an admin. If you're on a time crunch though you can leave this as is and ignore my comments about the middleware/decorator